src/core/declarativewebcontainer.cpp

```diff
@@ -519,6 +519,11 @@ bool DeclarativeWebContainer::eventFilter(QObject *obj, QEvent *event)
     if (event->type() == QEvent::Close && m_mozWindow) {
         m_mozWindow->suspendRendering();
         m_closing = true;
+
+        if (m_webPages->count() == 0) {
+            m_mozWindow.reset();
```

Comment:
Yes it is in use. Called when the last tab is closed.

I think that closeeventfilter should be merged to the webcobtainer/browser but didn't want to do that in this MR. All filtering is done now in webcontainer (only connections based op are left in closeeventfilter). Thus, "exit browser when downloading is finished" could be done differently. Case being when browser is "closed" before downloading is finished and "restarted" again when the same downloading is still ongoing. So, we also need to alter how the browser content window surface is recreated incase browser is closed while downloading is ongoing and restarted again.